### ISSUE TITLE
Publishing Test Fails on macOS Sierra

### ISSUE THREAD

**Issue by wjk** [dotnet/sdk#834]
I was trying to build [my fork of dotnet/sdk](https://github.com/wjk/dotnet-sdk) in Release configuration, but one of the tests failed. Specifically, the ["It builds a runnable output" test](https://github.com/dotnet/sdk/blob/master/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedApp.cs), when published on my Mac running macOS Sierra, does _not_ output a runnable entry point (apphost), nor does it write out the hostfxr or hostpolicy libraries. (The DLL, PDB, and the various JSON files are present, however.)

Some informal testing on the output of `dotnet new console` (modified only to add an explicit RID) reveals that a simple `dotnet build` does not write out the entry point or the host libraries.  While`dotnet publish` copies all the DLL dependencies, it does not add an entry point or the host libraries, either! This seems like a serious problem. Can anyone point out what might be going wrong? My platform information is below. Thanks!

```
.NET Command Line Tools (1.0.0-rc4-004767)

Product Information:
 Version:            1.0.0-rc4-004767
 Commit SHA-1 hash:  f7978d90e6

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.0.0-rc4-004767
```

**Comment by nguerrera** [comment]
@eerhardt

**Comment by wjk** [comment]
After doing some spelunking through the SDK targets files, as well as the output of `dotnet msbuild /nologo /t:Publish /p:RuntimeIdentifier=osx.10.12-x64 /v:diag`, I have determined the following:

* The native host, fxr, and other native files are not being copied into the output directory because the `_ComputeNETCoreBuildOutputFiles` target is not being run.
* This target is only run if the `ComputeNETCoreBuildOutputFiles` property is set to `true`.
* This variable is set [here](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets#L39). The condition on the property assignment statement indicates that it will only be set if the variable is unset at the time the property is evaluated, and the target framework identifier is equal to `.NETCoreApp`.
* MSBuild indicates that the `_ComputeNETCoreOutputFiles` variable is not set anywhere in the build. 
* MSBuild also indicates that `$(TargetFrameworkIdentifier)` is being set to `.NETCoreApp` in the "initial properties" phase of the build.

Therefore, all indications point to the target should be running. However, it still isn't. I don't currently have any other information as to what may be going wrong. The only thing I can think of is that the target framework identifier is being set too late in the build cycle, after it is already tested for. However, nowhere in the diagnostic output is it listed that the variable is ever being set after the "Initial properties" phase. Hope this helps!

**Comment by eerhardt** [comment]
Sorry, I've been working on a fix and didn't update this thread.

The issue is that `netcoreapp1.0` doesn't support RID `osx.10.12-x64`.  When we shipped `netcoreapp1.0`, OSX Sierra wasn't released yet, so we didn't support that RID. It only supports `osx.10.10-x64` and `osx.10.11-x64`.

`netcoreapp1.1` however, does support Sierra's RID.

Normally NuGet gives you warnings/errors when you try restoring for RIDs that are unsupported, but we recently removed those "guard rails". (/cc @emgarten).

The fix for the tests, and anyone's project who is targeting `netcoreapp1.0` is to either target `netcoreapp1.1` or use a lower RID, which is supported by `netcoreapp1.0`.

To fix the tests, I'm taking the latter approach - use `osx.10.11-x64`.
